logline_test.go

```diff
@@ -131,88 +133,436 @@ func TestLogLineRecordConformance(t *testing.T) {
 		}
 	})
 }
 
-func BenchmarkParse(b *testing.B) {
-	text := []byte("value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value,value")
-	b.Run("bench", func(b *testing.B) {
-		for n := 0; n < b.N; n++ {
-			ll := LogLine{FieldSeparator: 44}
-			ll.Parse(text, nil)
+func TestLogLineCustomFields(t *testing.T) {
+	t.Run("parse error to many fields", func(t *testing.T) {
+		values := make([]string, 0, LogLineNumFields)
+		for i := 0; i < int(LogLineNumFields)+1; i++ {
+			values = append(values, "value")
+		}
+		want := []byte(strings.Join(values, ","))
+
+		ll := LogLine{FieldSeparator: ','}
+		err := ll.Parse(want, nil)
+		if err != errLogLineTooManyFields {
+			t.Errorf("got: %s want: %s", err, errLogLineTooManyFields)
+		}
+	})
+
+	t.Run("set/get custom fields", func(t *testing.T) {
+		ll := LogLine{FieldSeparator: ','}
+
+		want := []byte("custom1")
+		idx := LogLineNumFields
+		ll.Set(idx, want)
+		got := ll.Get(idx)
+		if !bytes.Equal(got, want) {
+			t.Errorf("got: %s want: %s", got, want)
+
+		}
+
+		want = []byte("customN")
+		idx = LogLineNumFields + NumFieldsBaker - 1
+		ll.Set(idx, want)
+		got = ll.Get(idx)
+		if !bytes.Equal(got, want) {
+			t.Errorf("got: %s want: %s", got, want)
+		}
+
+		// Custom fields should not be serialized.
+		got = ll.ToText(nil)
+		if !bytes.Equal(got, []byte{}) {
+			t.Errorf("got: %s want: ''", got)
 		}
 	})
 }
 
-func BenchmarkToText(b *testing.B) {
-	b.Run("bench", func(b *testing.B) {
-		ll := LogLine{FieldSeparator: 44}
-		for i := 0; i < 100; i++ {
+func TestLogLineGet(t *testing.T) {
+	t.Run("no set value1", func(t *testing.T) {
+		ll := LogLine{FieldSeparator: ','}
+
+		for i := FieldIndex(0); i < LogLineNumFields+NumFieldsBaker; i++ {
+			got := ll.Get(i)
+			if got != nil {
+				t.Errorf("got: %s, want: ''", got)
```

Comment:
`got: %v, want: %v` is generally not always enough to make the failure message helpful and/or to identify without having to look at the line number. having `i` would be helpful here
```suggestion
				t.Errorf("ll.Get(%d) = %q, want nil", i, got)
```